packages/core/src/branch-assignment.ts

```
@@ -11,6 +11,68 @@ export interface EntryPointBranchAssignment {
+ * angle brackets, leading dash, refspec specials) are rejected.
+ */
+export function isValidBranchGroupBranchName(name: string): boolean {
+  if (typeof name !== "string") return false;
+  const trimmed = name.trim();
+  if (trimmed.length === 0) return false;
+  if (trimmed !== name) return false; // surrounding whitespace
+  if (name.length > 255) return false;
+  if (name.startsWith("-")) return false;
+  if (/\s/.test(name)) return false;
+  // Shell / refspec metacharacters that could escape a single git arg.
+  if (/[$`;|&<>(){}\[\]"'\\!*?~^:]/.test(name)) return false;
```

ESLint

[error] Unnecessary escape character: \[.

(no-useless-escape)